src/controllers/auth.js

```diff
@@ -0,0 +1,83 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import Users from '../models/Users';
+import Errors from '../helpers/errors';
+
+dotenv.config();
+
+/**
+ * Contains the authentification routes
+ *
+ * @class Auth
+ */
+class Auth {
+
+    /**
+   * Checks whether the user exists,
+   * if not, a new one is created,
+   * else the same user is returned
+   *
+   * @static
+   * @param {*} res
+   * @param {*} user
+   * @returns {object} user
+   * @memberof User
+   */
+    static async findOrCreate(res, user) {
+
+
+        try {
+            const {
+                id, email, picture, displayName,
+            } = user;
+
+            const username = displayName.replace(/\s+/g, '') + id.substr(0, 5);
+            const token = jwt.sign({ email, username }, process.env.SECRET, { expiresIn: '2d' });
+
+
+            const exist = await Users.findOne({ email });
+
+            if (exist == null) {
+
+                const result = await Users.create({
```

Comment:
@gracelungu , I think it could be better to make use of all information provided by the google/twitter/github to improve the user experience like firstname, lastname, etc.

Reply:
I have added the firstName and lastName from the provider information.